core/lib/dal/src/events_web3_dal.rs

```diff
@@ -185,7 +239,38 @@ impl EventsWeb3Dal<'_, '_> {
             )
             .fetch_all(self.storage.conn())
             .await?;
+
+            if api_mode == APIMode::Modern {
```

Comment:
Is it possible to extract shared code mapping logs between this method, `get_logs()` and `get_transaction_receipt`?